invenio_drafts_resources/resource_units.py

```diff
@@ -3,19 +3,24 @@
 # Copyright (C) 2020 CERN.
 # Copyright (C) 2020 Northwestern University.
 #
-# Invenio-Drafts-Resources is free software; you can redistribute it and/or
+# Invenio-Records-Resources is free software; you can redistribute it and/or
 # modify it under the terms of the MIT License; see LICENSE file for more
 # details.
 
-"""Resource Unit."""
+"""Resource units."""
 
+from invenio_records_resources.resource_units import IdentifiedRecord
 
-class IdentifiedDraft:
-    """Resource unit representing pid + Draft data clump."""
 
-    # TODO: IMPLEMENT ME!
-    def __init__(self, pid=None, draft=None):
-        """Initialize the draft state."""
-        self.id = pid.pid_value
+class IdentifiedRecordDraft(IdentifiedRecord):
+    """Resource unit representing pid + Record data clump.
+
+    Since a draft can be created for a non-existing record,
+    it might not have a PID.
+    """
+
+    def __init__(self, pid=None, record=None):
+        """Initialize the record state."""
+        self.id = pid.pid_value if pid else None
         self.pids = [pid]
```

Comment:
If pid is None you will end up with `[None]`. Is this intentional?

Reply:
Because of the above. Since we do not want to mint drafts, they do not have a PID if they are for a non-existing record. If the record exists, they do have a pid. Therefore `[None]` is okay when dealing with drafts.

Reply:
I was just thinking that if you have `pids=[None]` then any code that iterates through it, it must check `if pid is not None`. However, we could just check before if pid when we assign the value to attribute `self.pids`. An empty array would be more developer friendly :) WDYT?

Reply:
You are totally correct. Adding that in a separate PR :)